java-checks/src/main/java/org/sonar/java/checks/sustainability/AndroidExactAlarmCheck.java

```diff
@@ -73,7 +67,7 @@ private void onSetWindowInvocation(MethodInvocationTree tree) {
     var windowLengthMillisArg = tree.arguments().get(2);
     var windowLengthMillis = ExpressionUtils.resolveAsConstant(windowLengthMillisArg);
```

Comment:
Sorry, I just realized that all this can also be rewritten in a much simpler way:
```java
windowLengthMillisArg
  .asConstant(Long.class)
  .filter(value -> value < SUGGESTED_MIN_LENGTH_MILLIS)
  .ifPresent(value -> reportIfInAndroidContext(windowLengthMillisArg, "Use alarm windows of 10 minutes or more instead."));
```